ofrak_core/src/ofrak/core/lz4.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from dataclasses import dataclass
+
+import lz4.frame
+
+from ofrak.component.packer import Packer
+from ofrak.component.unpacker import Unpacker
+from ofrak.core.binary import GenericBinary
+from ofrak.core.magic import MagicDescriptionPattern, MagicMimePattern
+from ofrak.resource import Resource
+from ofrak_type.range import Range
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class Lz4Data(GenericBinary):
+    """
+    An LZ4 binary blob.
+
+    LZ4 is a high-speed lossless compression algorithm supporting multiple frame formats:
+    - Modern frame format (default)
+    - Legacy frame format
+    - Skippable frames
+    """
+
+
+class Lz4Unpacker(Unpacker[None]):
+    """
+    Unpack (decompress) an LZ4 file.
+
+    Supports all LZ4 frame formats:
+    - LZ4 default frame (modern format with metadata)
+    - LZ4 legacy frame (older format for backward compatibility)
+    - LZ4 skippable frames (metadata containers)
+    """
```

Comment:
Details about type of LZ4 compression need to be captured during unpacking and stored as attributes.

Reply:
Alternatively, is this information that can be gathered as part of identification?

Reply:
Added this in.